src/wrapper/jnienv.rs

```diff
@@ -298,6 +298,23 @@ impl<'a> JNIEnv<'a> {
         })
     }
 
+    /// Creates a new local reference frame, in which at least a given number of local
+    /// references can be created.
+    ///
+    /// Contrary to the `PushLocalFrame` from the `JNI` this function returns `Err` instead
+    /// of a negative value on failure.
```

Comment:
I'd add cross-references between `push_local_frame`, `pop_local_frame`, `AutoLocal` & `delete_local_ref` and update their API documentation, because all of them try to solve the same problem, but in a different way. 

@DarkEld3r , @jechas01 , Is that correct to give such recommendations:
 - Prefer `push_local_frame`/`pop_local_frame` if you create a _bounded_ number of local references in a method but can't rely on automatic de-allocation (e.g., in case of recursion or just deep call stacks).
 - Prefer `AutoLocal` in loops.
 - Avoid using `delete_local_reference` directly in favour of the first two approaches?